peru/main.py

```diff
@@ -221,6 +221,59 @@ def do_override(params):
             print('{}: {}'.format(module, params.runtime.get_override(module)))
 
 
+@peru_command('module', '''\
+Usage:
+    peru module add [<module>] [<url>] [<type>] [--rev=<rev>] [--reup=<reup>] 
+             [--filename=<filename>] [--sha=<sha>] [--unpack=<unpack>] [--yaml=<yaml>]
+    peru module --help
+
+Add a module to your Peru yaml file. Reguires a module name, url, and module type.
+
+Options:
+    add                       add new module
+      --rev=<rev>             set module rev value
+      --reup=<reup>           set module reup value
+      --filename=<filename>   set module filemame value
+      --sha=<sha>             set module sha1 value
+      --unpack=<unpack>       set module unpack value 
+      --yaml=<yaml>           file to use instead of peru.yaml     
+      --help 
+''')
+def do_add(params):
+    if params.args['add']:
+        if not params.args['<module>']:
+            params.runtime.display.print('Please specify a module to add. See usage with peru module --help')
+        elif not params.args['<url>']:
+            params.runtime.display.print('Please specify a module url. See usage with peru module --help')
+        elif not params.args['<type>']:
+            params.runtime.display.print('Please specify a module type. See usage with peru module --help')
+        else:
+            peru_file = params.args['--yaml'] if params.args['--yaml'] else 'peru.yaml'
+
+            if(peru_file[-5:] != '.yaml'):
+                params.runtime.display.print('Peru file should end in .yaml. See usage with peru add --help')
+
+            else:
+                with open(peru_file, "a") as yaml:
+                    yaml.write('\n' + params.args['<type>'] + ' module ' + params.args['<module>'] + ':')
+                    if(params.args['<url>']):
+                        yaml.write('\n  url: ' + params.args['<url>'])
+                    if(params.args['--rev']):
+                        yaml.write('\n  rev: ' + params.args['--rev'])
+                    if(params.args['--reup']):
+                        yaml.write('\n  reup: ' + params.args['--reup'])
+                    if(params.args['--filename']):
+                        yaml.write('\n  filename: ' + params.args['--filename'])
+                    if(params.args['--sha']):
+                        yaml.write('\n  sha: ' + params.args['--sha'])
+                    if(params.args['--unpack']):
+                        yaml.write('\n  unpack: ' + params.args['--unpack'])
```

Comment:
Kind of similar to what came up in https://github.com/buildinspace/peru/pull/165, because we support user-defined module types, we need to be prepared for modules with new field names that we don't know about yet. Hardcoding the field names as flags like this makes it hard to handle that case.